### ISSUE TITLE
YOLOv4-tiny is released

### ISSUE THREAD

**Issue by AlexeyAB** [hhk7734/tensorflow-yolov4#4]
@hhk7734 Hi,
Nice work with pypi: https://pypi.org/project/yolov4/

Did you manage to realize CIoU-loss and Mosaic?

Also YOLOv4-**tiny** is released:

* **source code**: https://github.com/AlexeyAB/darknet
* **cfg**: https://raw.githubusercontent.com/AlexeyAB/darknet/master/cfg/yolov4-tiny.cfg
* **weights**: https://github.com/AlexeyAB/darknet/releases/download/darknet_yolo_v4_pre/yolov4-tiny.weights
* **discussion**: https://github.com/AlexeyAB/darknet/issues/6067

There is required only 1 feature:

1. Add `groups=` and `group_id=` to the `[route]` layer.
```ini
[route]
layers=-1
groups=2
group_id=1
```

So if input is `WxHxC`, it divides input into 2 groups `WxHx(C/2)` (there are 2 groups: 0 and 1), and loads the 2nd group_1 `WxHx(C/2)`.

If there are many layers specified in `layers=` parameter, then this will be done for each of the input layers specified in `layer=`, then results will be concatenated across channels.

![cmp](https://user-images.githubusercontent.com/4096485/85734112-6e366700-b705-11ea-95d1-fcba0de76d72.png)

**Comment by hhk7734** [comment]
@AlexeyAB  Hey~! Hi :sunglasses:

**hunglc007/tensorflow-yolov4-tflite** has some problems with training.
After solving this problem, I will manage BoF functions.

Then, I try to add YOLOv4-tiny. Thanks for letting me know.

**Comment by AlexeyAB** [comment]
@hhk7734 Hi,

> hunglc007/tensorflow-yolov4-tflite has some problems with training.

* Did you solve this issue?


* Also how many FPS did you get for YOLO4-tiny on Coral board(TPU)? https://github.com/hhk7734/tensorflow-yolov4#objective

**Comment by hhk7734** [comment]
1. hunglc007 solved the problem, so I applied the solution.
1. Failed to compile the whole. Some sequence or operations are not compiled. The part that failed compilation was replaced by numpy. So, TPU consumes 55 ~ 60ms, the others are 35 ~ 40ms. FPS is 10~11.

**Comment by hhk7734** [comment]
CIoU
- utility: train: refactor bbox_ciou() 3a3d6522b5d59
- utility: train: move to tf.train 1bcce65e28156301
- tf: train: remove tf.math.divide_no_nan db12202497a091

Tiny
- model: yolov4: implement YOLOv4Tiny 65dcfdb33206c8710

Mosaic
- tf: dataset: implement mosaic  5c8cd15d1ccf66c0d

Requests were reflected. :)

**Comment by AlexeyAB** [comment]
@hhk7734 Great!

Do you know the reason? https://github.com/hunglc007/tensorflow-yolov4-tflite/issues/165

**Comment by hhk7734** [comment]
I studied artificial intelligence since May. So there are many things that I do not know.

**Comment by ankandrew** [comment]
@hhk7734 Very interesting to see v4 tiny on Edge TPU. I have two questions
1. What ops where not mapped to the TPU?
2. Did you quantize (post-training/training-aware) to INT8?

Thanks
